bench/tiered/push_pull.c

```diff
@@ -0,0 +1,306 @@
+    }
+
+    for (counter = 0; counter < MAX_RUN; ++counter) {
+        avg_time += avg_time_array[counter];
+        avg_throughput += avg_throughput_array[counter];
+        avg_file_size += avg_filesize_array[counter];
+    }
+
+    printf("Bytes Written- %" PRIi64
+           " (~%s), Time took- %.3f seconds, Throughput- %.3f MB/second\n",
+      avg_file_size / MAX_RUN, suffix, avg_time / MAX_RUN, avg_throughput / MAX_RUN);
+}
+
+/*
+ * run_test --
+ *     TODO: Add a comment describing this function.
+ */
+static void
+run_test(const char *home, uint64_t num_records, bool flush, int counter)
+{
+    struct timeval start, end;
+
+    char buf[1024];
+    double diff_sec;
+    int64_t file_size;
+
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    file_size = 0;
+    testutil_make_work_dir(home);
+    if (opts->tiered_storage) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/bucket", home));
+        testutil_make_work_dir(buf);
+    }
+
+    testutil_wiredtiger_open(opts, home, conn_config, NULL, &conn, false);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, opts->uri, table_config_row));
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), flush ? "flush_tier=(enabled,force=true)" : ""));
+
+    gettimeofday(&start, 0);
+    populate(session, num_records);
+
+    testutil_check(session->checkpoint(session, buf));
+    gettimeofday(&end, 0);
+
+    diff_sec = difftime_sec(start, end);
+
+    /* Sleep to guarantee the tables are created to read the size. */
+    sleep(1);
+
+    get_file_size(home, &file_size);
+    testutil_assert(diff_sec > 0);
+
+    avg_time_array[counter] = diff_sec;
+    avg_throughput_array[counter] = ((file_size / diff_sec) / WT_MEGABYTE);
+    avg_filesize_array[counter] = file_size;
+}
+
+/*
+ * populate --
+ *     Populate the table.
+ */
+static void
+populate(WT_SESSION *session, uint64_t num_records)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t i, str_len;
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + (uint32_t)__wt_random(&rnd) % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+    for (i = 0; i < num_records; i++) {
+        cursor->set_key(cursor, i + 1);
+        cursor->set_value(cursor, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+/*
+ * compute_tiered_file_size --
+ *     Iterate over all the tiered files and compute file size..
+ */
+static void
+compute_tiered_file_size(const char *home, const char *tablename, int64_t *file_size)
+{
+    char stat_path[512];
+    int index;
+    struct stat stats;
+
```

Comment:
`*file_size` should be initialized to 0 in this function. It's of course possible to require the caller initialize the value, as you have it now, but that's more unusual.  The idiom for WT, and most other C code, is that trailing parameters that are (type *) where the value is set in the function, should always have a value set in the function.  Easiest way to ensure that is initialize to 0.  Note this applies to `compute_tiered_file_size` and `compute_wt_file_size`, but not `get_file_size` (it does _not_ set `*file_size` in the function).

Reply:
Fixed